test-suite/testkit/src/api.rs

```diff
@@ -360,6 +423,51 @@ where
 
         Err(error)
     }
+
+    fn response_to_api_result_new<R>(mut response: Response) -> api::ApiResult<R>
+    where
+        R: DeserializeOwned + 'static,
+    {
+        fn extract_error(response: &mut Response) -> Option<api::ApiError> {
+            let code = response.status();
+            let body = response.text().expect("Unable to get response text");
+
+            let mut error = api::ApiError::new(code);
+            if let Some(json_obj) = serde_json::from_str::<serde_json::Value>(&body)
+                .unwrap()
+                .as_object()
+            {
+                if json_obj.contains_key("type") {
```

Comment:
Is there anything preventing implementing error parsing via `impl Deserialize for ApiError`? It would be a lot less verbose.

Reply:
It won't have an error code set. Not sure if this can be considered correct deserializing. Usually when you deserialize something, you expect to obtain a complete object.

Reply:
I'd think we can agree to go with an almost correct serialization in this case. Another way around is to split `ApiError` into body and code and deserialize the body:

```rust
struct ApiError {
    pub http_code: StatusCode,
    pub body: ApiErrorBody,
}

#[derive(Serialize, Deserialize)]
struct ApiErrorBody {
    // all fields except for `http_code`...
}
```

Reply:
Yeah, option with split body/code sounds good.